src/System.Management.Automation/engine/remoting/client/remoterunspace.cs

```diff
@@ -2005,27 +2005,213 @@ public override void StopProcessCommand()
             }
         }
 
-        /// <summary>
-        /// Adds the provided set of breakpoints to the debugger.
-        /// </summary>
-        /// <param name="breakpoints">Breakpoints.</param>
-        public override void SetBreakpoints(IEnumerable<Breakpoint> breakpoints)
-        {
-            _runspace.Debugger?.SetBreakpoints(breakpoints);
-        }
-
         /// <summary>
         /// Get a breakpoint by id, primarily for Enable/Disable/Remove-PSBreakpoint cmdlets.
         /// </summary>
         /// <param name="id">Id of the breakpoint you want.</param>
-        public override Breakpoint GetBreakpoint(int id) =>
-            _runspace.Debugger?.GetBreakpoint(id);
+        public override Breakpoint GetBreakpoint(int id)
+        {
+            // This is supported only for PowerShell versions >= 7.0
+            if (!_remoteDebuggingCapability.IsCommandSupported(RemoteDebuggingCommands.GetBreakpoint))
+            {
+                throw new PSNotSupportedException(
+                    StringUtil.Format(
+                        DebuggerStrings.CommandNotSupportedForRemoteUseInServerDebugger,
+                        RemoteDebuggingCommands.CleanCommandName(RemoteDebuggingCommands.GetBreakpoint)));
+            }
+
+            return InvokeRemoteBreakpointFunction<Breakpoint>(
+                RemoteDebuggingCommands.GetBreakpoint,
+                new Dictionary<string, object>
+                {
+                    { "Id", id },
+                });
+        }
 
         /// <summary>
         /// Returns breakpoints primarily for the Get-PSBreakpoint cmdlet.
         /// </summary>
-        public override List<Breakpoint> GetBreakpoints() =>
-            _runspace.Debugger?.GetBreakpoints();
+        public override List<Breakpoint> GetBreakpoints()
+        {
+            // This is supported only for PowerShell versions >= 7.0
+            if (!_remoteDebuggingCapability.IsCommandSupported(RemoteDebuggingCommands.GetBreakpoint))
+            {
+                throw new PSNotSupportedException(
+                    StringUtil.Format(
+                        DebuggerStrings.CommandNotSupportedForRemoteUseInServerDebugger,
+                        RemoteDebuggingCommands.CleanCommandName(RemoteDebuggingCommands.GetBreakpoint)));
+            }
+
+            CheckForValidateState();
+
+            var breakpoints = new List<Breakpoint>();
+
+            using (PowerShell ps = GetNestedPowerShell())
+            {
+                ps.AddCommand(RemoteDebuggingCommands.GetBreakpoint);
+
+                Collection<PSObject> output = ps.Invoke<PSObject>();
+                foreach (var item in output)
+                {
+                    if (item?.BaseObject is Breakpoint)
```

Comment:
```suggestion
                    if (item?.BaseObject is Breakpoint breakpointItem)
```
then use `breakpointItem`  for the `breakpoints.Add(breakpointItem)`

Reply:
I didn't know you could do that with `is`. Thanks! Will update in next commit.